hledger/Hledger/Cli/Commands/Register.hs

```diff
@@ -190,12 +191,16 @@ postingsReportItemAsText opts preferredamtwidth preferredbalwidth (mdate, mendda
             _                      -> (id,acctwidth)
     amt = showAmountsLinesB dopts . (\x -> if null x then [nullamt] else x) . amounts $ pamount p
     bal = showAmountsLinesB dopts $ amounts b
-    dopts = noPrice{displayColour=color_ . rsOpts $ reportspec_ opts}
+    dopts = registerOpts opts
     -- Since this will usually be called with the knot tied between this(amt|bal)width and
     -- preferred(amt|bal)width, make sure the former do not depend on the latter to avoid loops.
     thisamtwidth = maximumDef 0 $ map wbWidth amt
     thisbalwidth = maximumDef 0 $ map wbWidth bal
 
+registerOpts :: CliOpts -> AmountDisplayOpts
```

Comment:
What's the purpose of this ?

Reply:
This, along with the changes in Balance.hs, was an attempt to reduce code duplication. I'm not altogether satisfied with the result, so perhaps another approach would be better.